docs/developers/deprecating.md

```diff
@@ -0,0 +1,45 @@
+ 4. If a feature is shown to be worth keeping after it has been deprecated and before it has been removed, it can be de-deprecated in any minor version.
+
+These also apply to releases of the iCommands package because they are strictly tied to releases of the server.
+
+## How do I deprecate a feature?
+
+We will now describe the steps which should be taken to deprecate various types of features. In general, we seek to avoid changing or removing runtime functionality.
+
+The following things need to be done whenever a feature is being declared as deprecated:
+
+ 1. Update documentation to include alternative, supported way(s) of accomplishing what is now deprecated. If no alternative is available, include a note indicating that use of the deprecated API should be avoided. Updating the documentation includes adding a `\deprecated` tag with the aforementioned notes to the Doxygen for any public declarations associated with the feature.
+ 2. Add a "deprecated" attribute specifier (see [https://en.cppreference.com/w/cpp/language/attributes/deprecated](https://en.cppreference.com/w/cpp/language/attributes/deprecated)) to the C/C++ public declarations specific to this API and include a message that is similar if not identical to the message used in (1). This includes any functions, structs, classes, and typedefs. If a struct, class, or typedef is being used by other APIs which are not being deprecated and the usage is not related to the feature which is being deprecated, do not deprecate these. For example, `DataObjInp` could be used by an API which is being deprecated, but `DataObjInp` is not specific to that API. Therefore, `DataObjInp` would not be a candidate for deprecation.
+ 3. Add comments to any macros directly related to the API indicating that the feature has been deprecated. This will not have any effect on compilation or usage, but will give the developer pause as they consider whether to use it in an application.
+ 4. If the feature is used by a client such as the iCommands and has user-facing help text, a note about the feature being deprecated should be included in the help text anywhere the feature is mentioned. If the feature being deprecated *is* an iCommand, a note about its deprecation can be included near the top of the help text.
+
+The following rules should also be followed when deprecating an API:
+ - The API should NOT add a deprecation message to the RcComm or RsComm's rError object.
+ - The API should NOT emit a deprecation message to the log.
+
+## How do I remove a feature?
+
+Once a feature has been deprecated, it can be removed in the next major release. Please use caution when removing code, as clients and plugins may be depending on it. Be sure to consider this before releasing with removed features.
+
+### Removing APIs
+
+!!!
+TODO: maybe remove this section...
+!!!
```

Comment:
i think it's important to keep this section.

Reply:
Okay. I'll remove these 3 lines. Still open to wording / content suggestions

Reply:
Done